controllers/webhooks/component_webhook.go

```diff
@@ -56,9 +58,43 @@ func (w *ComponentWebhook) Register(mgr ctrl.Manager, log *logr.Logger) error {
 // +kubebuilder:webhook:path=/mutate-appstudio-redhat-com-v1alpha1-component,mutating=true,failurePolicy=fail,sideEffects=None,groups=appstudio.redhat.com,resources=components,verbs=create;update,versions=v1alpha1,name=mcomponent.kb.io,admissionReviewVersions=v1
 
 // Default implements webhook.Defaulter so a webhook will be registered for the type
+// Default retrieves the list of components that the Component nudges and updates their statuses to list
+// the component as a nudging component (status.BuildNudgedBy)
 func (r *ComponentWebhook) Default(ctx context.Context, obj runtime.Object) error {
+	return nil
+}
+
+// UpdateNudgedComponentStatus retrieves the list of components that the Component nudges and updates their statuses to list
+// the component as a nudging component (status.BuildNudgedBy)
+func (r *ComponentWebhook) UpdateNudgedComponentStatus(ctx context.Context, obj runtime.Object) error {
+	comp := obj.(*appstudiov1alpha1.Component)
+	compName := comp.Name
+
+	// For each component that the Component nudges, retrieve its resource and update its status accordingly
+	for _, nudgedCompName := range comp.Spec.BuildNudgesRef {
+		// Retrieved the nudged component
+		nudgedComp := &appstudiov1alpha1.Component{}
+		err := r.client.Get(ctx, types.NamespacedName{Namespace: comp.Namespace, Name: nudgedCompName}, nudgedComp)
+		if err != nil {
+			// Return an error if an error was encountered retrieving the resource.
+			// If the resource wasn't found yet - leave it however
```

Comment:
Ah, good catch. We should actually call `continue` here when it's not found and add a small log statement. 

Because k8s resources may be created not necessarily in order, we don't want to error out here, but rather let the error sort itself out after all resources have been created

Reply:
What are the repurcussions if the status doesnt get set for a Component that is not updated

Reply:
In this scenario, The status will be set when the missing component gets created.

Reply:
So wouldnt that require the current Component to be triggered for an update again to go through this webhook logic?

Reply:
But even then, if the status doesn’t get set, there shouldn’t be any repurcussions, as that’s just a status field we’re consuming to manage the relationship. The build nudges ref consumers don’t use it (they only use the spec.BuildNudgesRef)

Reply:
So my concern is if CompA nudges CompB:

1. CompA webhook create - go through the nudge ref, try to get CompB.
2. CompB is present, fine. Update its status.
3. CompB is absent, log and continue.
4. CompA is updated, webhook update - go through the nudge ref, try to get CompB.
5. CompB is present, fine. Update its status.

However, if  Step 4 never happes ie CompA is never updated, CompB status wont be updated as well?

Reply:
Right, I see now.

 So the status may not get updated in that case, but since this status isn’t used by the build nudges ref consumers (i.e. build service), I think that’s a fine trade off.